test-clients/src/main/java/com/hedera/services/bdd/suites/schedule/ScheduleSignSpecs.java

```diff
@@ -66,10 +65,105 @@ public static void main(String... args) {
 						scheduleSigIrrelevantToSchedulingTxn(),
 						overlappingKeysTreatedAsExpected(),
 						retestsActivationOnSignWithEmptySigMap(),
+						basicSignatureCollectionWorks(), // 19
+						addingSignaturesToExecutedTxFails(), // 20
+						addingSignaturesToNonExistingTxFails(), // 21
+//						addingInvalidEd25519Fails(), // 22
+//						addingEd25519WithInvalidFormatFails(), // 23
+						addingSignatureByNonRequiredSignerFails(), // 24
+						addingSignatureByNonRequiredSignerFails2(), // 25
+
 				}
 		);
 	}
 
+	private HapiApiSpec basicSignatureCollectionWorks() {
+		var txnBody = cryptoTransfer(tinyBarsFromTo("sender", "receiver", 1)).signedBy("somebody");
+
+		return defaultHapiSpec("BasicSignatureCollectionWorks")
+				.given(
+						cryptoCreate("sender"),
+						cryptoCreate("receiver"),
+						newKeyNamed("somebody"),
+						scheduleCreate("basicXfer", txnBody)
+				)
+				.when(
+						scheduleSign("basicXfer").withSignatories("somebody", "sender")
+				)
+				.then(
+						getScheduleInfo("basicXfer")
+								.hasSignatories("somebody", "sender")
+				);
+	}
+
+	private HapiApiSpec addingSignatureByNonRequiredSignerFails() {
+		var txnBody = cryptoTransfer(tinyBarsFromTo("sender", "receiver", 1)).signedBy("somebody");
+
+		return defaultHapiSpec("AddingSignatureByNonRequiredSignerFails")
+				.given(
+						cryptoCreate("sender"),
+						cryptoCreate("receiver"),
+						newKeyNamed("somebody"),
+						newKeyNamed("somebodyelse"),
+						scheduleCreate("basicXfer", txnBody)
+				)
+				.when(
+						scheduleSign("basicXfer").withSignatories("somebodyelse").hasKnownStatus(SOME_SIGNATURES_WERE_INVALID)
+				)
+				.then(
+				);
+	}
+
+	private HapiApiSpec addingSignatureByNonRequiredSignerFails2() {
+		var txnBody = mintToken("tokenA", 50000000L);
+
+		return defaultHapiSpec("AddingSignatureByNonRequiredSignerFails")
+				.given(
+						cryptoCreate("sender"),
+						cryptoCreate("receiver"),
+						newKeyNamed("somebody"),
+						newKeyNamed("admin"),
+						newKeyNamed("mint"),
+						newKeyNamed("newMint"),
+						tokenCreate("tokenA").adminKey("admin").supplyKey("mint"),
+						scheduleCreate("tokenMintScheduled", txnBody)
+				)
+				.when(
+						tokenUpdate("tokenA").supplyKey("newMint"),
+						scheduleSign("tokenMintScheduled").withSignatories("mint").hasKnownStatus(SOME_SIGNATURES_WERE_INVALID)
+				)
+				.then(
+				);
+	}
+
+	private HapiApiSpec addingSignaturesToNonExistingTxFails() {
```

Comment:
This test fails it cannot find a corresponding tx body for `non-existing`. More like bad `HapiScheduleSign` implementation